core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -0,0 +1,86 @@
+package com.sequenceiq.cloudbreak.service;
+
+import com.sequenceiq.cloudbreak.domain.stack.loadbalancer.TargetGroup;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.knox.KnoxRoles;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceGroup;
+import com.sequenceiq.cloudbreak.template.model.ServiceComponent;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+import com.sequenceiq.common.api.type.LoadBalancerType;
+
+@Service
+public class LoadBalancerConfigService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoadBalancerConfigService.class);
+
+    private static final String ENDPOINT_SUFFIX = "gateway";
+
+    private static final String PUBLIC_SUFFIX = "external";
```

Comment:
I think the convention we should use is to follow AWS terminology. internet-facing load balancers do not have an "external" suffix, but internal load balancers have and "-internal" suffix.

Reply:
I'm going to copy this comment to CB-9900. External subnets aren't going to be incorporated until that's done, so we can look at the naming then.